src/hooks/programPlan.ts

```diff
@@ -0,0 +1,195 @@
+import { gql, useMutation, useQuery } from '@apollo/client'
+import { useApp } from 'lodestar-app-element/src/contexts/AppContext'
+import hasura from '../hasura'
+import { CardProducts, MemberShipCards } from '../types/programPlan'
+
+export const useMembershipCardByAppId = () => {
+  const { id: appId } = useApp()
+  const { data: cardData } = useQuery<hasura.GetCardTitleByAppId, hasura.GetCardTitleByAppIdVariables>(
+    gql`
+      query GetCardTitleByAppId($appId: String) {
+        card(where: { app_id: { _eq: $appId } }) {
+          title
+          id
+        }
+      }
+    `,
+    {
+      variables: { appId },
+    },
+  )
+
+  const membershipCards: MemberShipCards = (cardData && cardData.card.map(card => card)) || []
+
+  return {
+    membershipCards,
+  }
+}
+
+export const useMembershipCardByTargetId = (productType: string, targetId: string) => {
+  const { data: cardData, refetch: refetchMembershipCard } = useQuery<
+    hasura.GetCardByTargetId,
+    hasura.GetCardByTargetIdVariables
+  >(
+    gql`
+      query GetCardByTargetId($productType: String!, $targetId: uuid!) {
+        card_product(where: { product_type: { _eq: $productType }, target: { _eq: $targetId } }) {
+          id
+          target
+          product_type
+          card_id
+          card {
+            id
+            title
+          }
+        }
+      }
+    `,
+    {
+      variables: { productType, targetId },
+    },
+  )
+  const cardTitle: string =
+    (cardData && cardData.card_product.map(cardProduct => cardProduct.card.title).toString()) || ''
```

Comment:
1. I agree with you, using a ternary operator can avoid guessing by reader. Adjusted.
2. This is a deliberate design by me, because I consider that in the future there may be a program plan with multiple membership cards, but don’t worry, currently the general user can only use the front-end to make one program plan with one membership card. So normal conditions will not be multiple membership card names for now.

Reply:
(cardData && cardData.card_product.map(cardProduct => cardProduct.card.title).toString()) || ''
    cardData ? cardData.card_product.map(cardProduct => cardProduct.card.title).toString() : ''
    cardData?.card_product.map(cardProduct => cardProduct.card.title).toString() || ''   // falsy
    cardData?.card_product.map(cardProduct => cardProduct.card.title).toString() ?? ''   // nullsy